bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.androiddebugbridge.internal.discovery;
+
+import static org.openhab.binding.androiddebugbridge.internal.AndroidDebugBridgeBindingConstants.*;
+
+import java.util.Dictionary;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AndroidTVMDNSDiscoveryParticipant} is responsible for discovering new and removed Android TV devices. It
+ * uses
+ * the central {@link MDNSDiscoveryService}.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@Component(configurationPid = "discovery.androiddebugbridge")
```

Comment:
Some bindings also set `service = MDNSDiscoveryParticipant.class` but not all.
Unfortunately, I don't know what is the recommendation.
So this is just a comment, not a request for change.

Reply:
Maybe @wborn or @jlaur you know what is the recommendation here ?

Reply:
I have added it to both services as seems appropriate.

Reply:
Also @fwolter

Reply:
If a `@Component` only implements one interface, bnd automatically adds that interface to the provided interfaces in the generated component XML files, e.g. `target/classes/OSGI-INF/org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant.xml `:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.3.0" name="org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant" activate="activate" modified="modified" configuration-pid="discovery.androiddebugbridge">
  <service>
    <provide interface="org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant"/>
  </service>
  <implementation class="org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant"/>
</scr:component>
```

So it is redundant if the list of `<provide interface=`  values is still correct if you remove the service from the `@Component`  annotation.

Sometimes the generated XML is wrong e.g. if your discovery service implements multiple interfaces or extends from the `AbstractDiscoveryService`... in that case it is mandatory to specify the list of service interfaces in the `@Component` annotation in order for openHAB Core to be able to use all implemented inferfaces.

Reply:
Thank you @wborn for the clear explanation.

So it was not required to add it in the current case but this is also not an error to have it.